deploy/templates/azuredeploy.standard.json

```diff
@@ -385,88 +411,29 @@
-            },
-            "dependsOn": [
-                "[variables('eventHubNamespaceResourceId')]"
-            ]
-        },
-        {
-            "comments": "CDM Telemetry Consumer Group of Event Hub for Device Telemetry",
+            "comments": "CDM Consumer Group for Device Telemetry EventHub",
             "type": "Microsoft.EventHub/namespaces/eventhubs/consumergroups",
-            "name": "[concat(parameters('eventHubNamespaceName'), '/', parameters('eventHubName'), '/', variables('eventHubConsumerGroup1'))]",
+            "name": "[concat(parameters('eventHubNamespaceName'), '/', parameters('eventHubName'), '/', variables('datalakeEventHubConsumerGroup'))]",
             "apiVersion": "2017-04-01",
             "properties": {
                 "userMetadata": "CDM Telemetry Consumer Group"
             },
             "dependsOn": [
-                "[variables('eventHubResourceId')]"
-            ]
-        },
-        {
-            "comments": "UX Telemetry Consumer Group of Event Hub for Device Telemetry",
-            "type": "Microsoft.EventHub/namespaces/eventhubs/consumergroups",
-            "apiVersion": "2017-04-01",
-            "name": "[concat(parameters('eventHubNamespaceName'), '/', parameters('eventHubName'), '/', variables('eventHubConsumerGroup2'))]",
-            "properties": {
-                "userMetadata": "UX Telemetry Consumer Group"
-            },
-            "dependsOn": [
-                "[variables('eventHubResourceId')]"
             ]
         },
         {
-            "comments": "Event Hub Authorization for Device Telemetry Consumers",
-            "type": "Microsoft.EventHub/namespaces/eventhubs/authorizationRules",
-            "name": "[concat(parameters('eventHubNamespaceName'), '/', parameters('eventHubName'), '/', parameters('eventHubAuthorizationName'))]",
-            "apiVersion": "2017-04-01",
-            "location": "[resourceGroup().location]",
+            "comments": "Datalake service principal role assignment",
+            "type": "Microsoft.Storage/storageAccounts/providers/roleAssignments",
+            "condition": "[not(empty(parameters('datalakeServicePrincipalId')))]",
+            "apiVersion": "2018-09-01-preview",
+            "name": "[concat(parameters('datalakeName'), '/Microsoft.Authorization/', guid(uniqueString(parameters('datalakeName'))))]",
             "properties": {
-                "rights": [
-                    "Listen"
-                ]
+                "roleDefinitionId": "[concat('/subscriptions/', subscription().subscriptionId, '/providers/Microsoft.Authorization/roleDefinitions/', 'ba92f5b4-2d11-453d-a403-e96b0029c9fe')]",
+                "principalId": "[parameters('datalakeServicePrincipalId')]",
+                "principalType": "ServicePrincipal"
             },
             "dependsOn": [
-                "[variables('eventHubResourceId')]"
+                "[variables('datalakeResourceId')]"
             ]
         },
```

Comment:
do we still need this if we changed the  adls storage to use connection string?

Reply:
No, technically not - however I want to keep it around so we can permission a user principal to it, just like we are now doing with the keyvault resource.